Realm/RLMObjectBase.mm

```diff
@@ -92,7 +92,8 @@ - (instancetype)initWithValue:(id)value schema:(RLMSchema *)schema {
             @throw RLMException(@"Invalid array input. Number of array elements does not match number of properties.");
         }
         for (NSUInteger i = 0; i < array.count; i++) {
-            [self setValue:RLMValidatedObjectForProperty(array[i], properties[i], schema) forKeyPath:[properties[i] name]];
+            id propertyValue = RLMValidatedObjectForProperty(array[i], properties[i], schema);
```

Comment:
is there a reason you're not calling `RLMNSNullToNil()` as part of `RLMValidatedObjectForProperty()`?